src/java.desktop/share/classes/sun/java2d/pipe/RenderBuffer.java

```diff
@@ -25,57 +25,57 @@
 
 package sun.java2d.pipe;
 
-import jdk.internal.misc.Unsafe;
+import java.lang.foreign.Arena;
+import java.lang.foreign.MemorySegment;
+import java.lang.foreign.ValueLayout;
+import static java.lang.foreign.ValueLayout.*;
 
 
 /**
- * The RenderBuffer class is a simplified, high-performance, Unsafe wrapper
+ * The RenderBuffer class is a simplified, high-performance class
  * used for buffering rendering operations in a single-threaded rendering
- * environment.  It's functionality is similar to the ByteBuffer and related
+ * environment.  Its functionality is similar to the ByteBuffer and related
  * NIO classes.  However, the methods in this class perform little to no
  * alignment or bounds checks for performance reasons.  Therefore, it is
  * the caller's responsibility to ensure that all put() calls are properly
  * aligned and within bounds:
  *   - int and float values must be aligned on 4-byte boundaries
  *   - long and double values must be aligned on 8-byte boundaries
+ * Failure to do so will result in exceptions from the FFM API, or worse.
  *
  * This class only includes the bare minimum of methods to support
  * single-threaded rendering.  For example, there is no put(double[]) method
  * because we currently have no need for such a method in the STR classes.
  */
-public class RenderBuffer {
+public final class RenderBuffer {
 
     /**
      * These constants represent the size of various data types (in bytes).
      */
-    protected static final long SIZEOF_BYTE   = 1L;
-    protected static final long SIZEOF_SHORT  = 2L;
-    protected static final long SIZEOF_INT    = 4L;
-    protected static final long SIZEOF_FLOAT  = 4L;
-    protected static final long SIZEOF_LONG   = 8L;
-    protected static final long SIZEOF_DOUBLE = 8L;
+    private static final int SIZEOF_BYTE   = 1;
+    private static final int SIZEOF_SHORT  = 2;
+    private static final int SIZEOF_INT    = 4;
+    private static final int SIZEOF_FLOAT  = 4;
+    private static final int SIZEOF_LONG   = 8;
+    private static final int SIZEOF_DOUBLE = 8;
```

Comment:
These fields could be eliminated and replaced with e.g. `JAVA_LONG.byteSize()` . I do not think the `int` vs. `long` matters that much if at all.